### ISSUE TITLE
FeatureFinderMultiplex detects too many peptide singlets for knock_out = true

### ISSUE THREAD

**Issue by MaxEsefeld** [OpenMS/OpenMS#3881]
Hi, 

I tested some dimethylated samples (+28 Da vs. +32 Da) with the new FeatureFinderMultiplex (version 2018.10.26 in KNIME). 
For me it seems as the new FFM behave strange when compared to the previous FFM 2.3 (TOPPAS version 2.3; 2018.04.27) 
The new consensusXML is bigger, but seems to contain less dimethylated pairs then the previous one. 
I discovered this problems already for the other nightly builds, 2018.08.22. and 2018.08.28 (the one on the USB-stick, we got at the Berlin meeting?)
Is there changed much in the code of the FFM or is it an user error? 
You will find the KNIME wf here: https://1drv.ms/f/s!AiPYIbX_1fMQgbERg_yX_t_WTGrDCQ
Also attached is the fasta, a mzML file, the mzTAB output and the old TOPPAS 2.3. 

Thanks
Max

**Comment by timosachsenberg** [comment]
Hi @lars20070 would you be so kind and check that?

**Comment by lars20070** [comment]
Yep, will do.

@MaxEsefeld Thanks for the data and testing.

**Comment by MaxEsefeld** [comment]
I haven´t test it yet, but maybe it depends on the Knock-out models. I gonna test if the option has to turned off, laterly.

**Comment by lars20070** [comment]
@MaxEsefeld Let's discuss results here. https://gitter.im/OpenMS/FeatureDetection

**Comment by lars20070** [comment]
In both cases (2.3 and 2.4), you used the same setting `algorithm:knock_out = true`?

**Comment by MaxEsefeld** [comment]
Yes I did

**Comment by lars20070** [comment]
Yes, `averagine_similarity_scaling` has no effect. PR with fix coming up.

@MaxEsefeld Thanks a lot for spotting this bug.

**Comment by lars20070** [comment]
@MaxEsefeld Please let me know if this fixes the problem.

**Comment by lars20070** [comment]
I am using a dimethyl-labelled, fixed-ratio 1:1 dataset to test the singlet detection. In such a dataset we expect only peptide pairs, i.e. we do not want to see any peptide singelts at all.

Contrary to this expectation, I do see very clear high-intensity, identified peptide singlets. @MaxEsefeld Do you know of any explanation?

**Comment by lars20070** [comment]
![screenshot from 2018-12-17 16-36-23](https://user-images.githubusercontent.com/6149805/50097367-0fdc1e80-021a-11e9-99d3-b22d82edca75.png)
![screenshot from 2018-12-17 16-36-46](https://user-images.githubusercontent.com/6149805/50097368-11a5e200-021a-11e9-8266-d02f40a14e82.png)

`(Dimethyl)EYSIEEILR` with 1% FDR was identified.

**Comment by lars20070** [comment]
![singlet](https://user-images.githubusercontent.com/6149805/50099252-f3da7c00-021d-11e9-82c7-bab498e6bbce.png)

Here another example in a single spectrum. There is no heavy partner at `613.8 Th`. These peaks clearly belong to the `612.3 Th` feature.

**Comment by MaxEsefeld** [comment]
> I am using a dimethyl-labelled, fixed-ratio 1:1 dataset to test the singlet detection. In such a dataset we expect only peptide pairs, i.e. we do not want to see any peptide singelts at all.
> 
> Contrary to this expectation, I do see very clear high-intensity, identified peptide singlets.
> @MaxEsefeld Do you know of any explanation?

@lars20070 
This might be caused by the preparation of the sample. 
It wasn´t expected to gain a ratio of 1:1 dimethyl labeled peptides. 

Normally to create a sample with fixed ratios one would split a sample into half. Then each of the two samples would be labeled with either light or heavy formaldehyde. After labeling the two samples would be merged and analyzed by MS/MS. 
But:
To avoid loss of time, I used two biologic identical samples [sample (A)XY and sample (B)XY] which were already labeled and merged. 
X and Y are two different states of the same origin of cells, 
e.g. X=blood cells with proteases and Y=blood cells without proteases. 
In short: 
Sample (A)XY were labeled in this fashion: X is the light channel and Y is the heavy channel. 

The Sample (B)XY were labeled in this fashion: X is the heavy channel and Y is this time the light channel. 

Then the I mixed 50% Sample A with 50% Sample B. 

Only the number of the light and the heavy labeled peptides should be near identical, but technically singlets were possible.

**Comment by lars20070** [comment]
[170811_OV4_LS_Blood_3_WP_H.pdf](https://github.com/OpenMS/OpenMS/files/2694590/170811_OV4_LS_Blood_3_WP_H.pdf)

I now find about 4-5% singlets in your dataset (both in the case of all peptide features and in the case of peptide features with mapped ID only). I checked some of these singlets manually and cannot find any light/heavy partner.

Can we close the issue?
